pkg/apis/pipeline/v1beta1/results.go

```diff
@@ -0,0 +1,26 @@
+/*
+Copyright 2022 The Tekton Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+// ResultsType indicates the type of a result;
+// Used to distinguish between a single string and an array of strings.
+type ResultsType string
```

Comment:
Maybe it should start having an interface for ResultType and ParamsType, as they ultimately need to be in ArrayOrString: https://github.com/tektoncd/pipeline/blob/983dc37c5de9dee2e7790d7aea48312f65a2e698/pkg/apis/pipeline/v1beta1/param_types.go#L106

Pinging others @dibyom @vdemeester for more opinions

Reply:
Any reason for not moving them to `ArrayOrString` "now", as, from the API perspective (consumption) it doesn't change too much, all current API usage will still work because `string` is still supported.

Reply:
Good point! I would suggest to add a TODO for changing this to have ResultType also in scope